drivers/storage/portworx/util/util.go

```diff
@@ -236,7 +236,7 @@ func IsPortworxEnabled(cluster *corev1.StorageCluster) bool {
 
 // IsCSIEnabled returns true if CSI is not disabled by the feature flag
 func IsCSIEnabled(cluster *corev1.StorageCluster) bool {
-	return IsPortworxEnabled(cluster) && FeatureCSI.IsEnabled(cluster.Spec.FeatureGates)
+	return IsPortworxEnabled(cluster) && cluster.Spec.CSI != nil && cluster.Spec.CSI.Enabled
```

Comment:
if we only remove feature gate on new installations then we shall check the csi flag here as well for existing installations.

Reply:
ah good point.. maybe the tradeoff is better to always move from feature gate -> CSI spec.